src/engine/Sound.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <unordered_map>
+#include "Buffer.h"
+
+
+class Sound
+{
+public:
+	Sound();
+	~Sound();
+	void play(std::string);
+	void stop(std::string);
+	void pause(std::string);
+	std::string createSoundBuffers(std::string, std::string, bool);
+	std::string InitializeSound();
+
+
+private:
+	//OpenAl needs to know to play the sound is the position and velocity of the
+	//source and the listener
+	//orientation of the listener is described by two vectors, one pointing in the direction the listener
+	//looks at and a second pointing up
+
+	
+	ALfloat SourcePos[3] = { 0.0, 0.0, 0.0 };
+	ALfloat SourceVel[3] = { 0.0, 0.0, 0.0 };
+	ALfloat ListenerPos[3] = { 0.0, 0.0, 0.0 };
+	ALfloat ListenerVel[3] = { 0.0, 0.0, 0.0 };
+	ALfloat ListenerOri[6] = { 0.0, 0.0, -1.0, 0.0, 1.0, 0.0 };
+
+	ALCcontext *context;
+	ALCdevice *device;
+
+	ALuint buffers[32];
```

Comment:
Why 32? Is this a limit imposed by OpenAL? If not, the resource manager can take care of generating buffers as needed.